### ISSUE TITLE
Muon never zeroes the gradients it delegates to its inner AdamW

### ISSUE THREAD

**Issue by Naeemkh** [KempnerInstitute/KempnerForge#182]
Muon splits its parameters at [`training/optimizer.py:508-511`](https://github.com/KempnerInstitute/KempnerForge/blob/17248f87347b4793727c6ea9aaac4fda103e8d21/kempnerforge/training/optimizer.py#L508), inside _build_muon: anything _is_muon_eligible accepts goes to Muon, the rest to an internal torch.optim.AdamW. Only the Muon half is then registered with the [base class](https://github.com/KempnerInstitute/KempnerForge/blob/17248f87347b4793727c6ea9aaac4fda103e8d21/kempnerforge/training/optimizer.py#L395):

          super().__init__(muon_params, defaults)

`torch.optim.Optimizer.zero_grad` walks `self.param_groups`, which therefore holds only the Muon half. It has no way to know self._adam exists. Muon bridges that gap everywhere the delegation was written out by hand, and `zero_grad` is the one place it was not.
